File: src/test/groovy/com/benjaminsproule/swagger/gradleplugin/test/jaxrs/ExtendedTestResourceWithoutClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.jaxrs;

import javax.ws.rs.GET;
import javax.ws.rs.Path;

import io.swagger.annotations.ApiOperation;

public class ExtendedTestResourceWithoutClassAnnotation extends TestResourceWithoutClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/root/withoutannotation/overriden")
    @GET
    @Override
    public String overriden() {
        return "";
    }
}

File: kotlin-tests/src/main/kotlin/com/benjaminsproule/swagger/gradleplugin/test/kotlin/jaxrs/ExtendedTestResourceWithoutClassAnnotation.kt
package com.benjaminsproule.swagger.gradleplugin.test.kotlin.jaxrs

import io.swagger.annotations.ApiOperation

import javax.ws.rs.GET
import javax.ws.rs.Path

class ExtendedTestResourceWithoutClassAnnotation : TestResourceWithoutClassAnnotation() {

    @ApiOperation("An overriden operation description")
    @Path("/root/withoutannotation/overriden")
    @GET
    override fun overriden(): String {
        return ""
    }

    @Path("/root/withoutannotation/overridenWithoutDescription")
    @GET
    override fun overridenWithoutDescription(): String {
        return ""
    }
}
